packages/graphql/src/translate/translate-update.ts

```diff
@@ -338,8 +338,18 @@ export default async function translateUpdate({
                     const relTypeStr = `[${relationVarName}:${relationField.type}]`;
 
                     if (!relationField.typeMeta.array) {
-                        const validateRelationshipExistance = `CALL apoc.util.validate(EXISTS((${varName})${inStr}[:${relationField.type}]${outStr}(:${refNode.name})),'Relationship field "%s.%s" cannot have more than one node linked',["${relationField.connectionPrefix}","${relationField.fieldName}"])`;
-                        createStrs.push(validateRelationshipExistance);
+                        const singleCardinalityValidationTemplate = (nodeName) =>
+                            `CALL apoc.util.validate(EXISTS((${varName})${inStr}[:${relationField.type}]${outStr}(:${nodeName})),'Relationship field "%s.%s" cannot have more than one node linked',["${relationField.connectionPrefix}","${relationField.fieldName}"])`;
+                        if (relationField.union) {
+                            const validateRelationshipExistance = relationField.union.nodes?.map(singleCardinalityValidationTemplate) as string[];
+                            createStrs.push(...validateRelationshipExistance);
+                        } else if (relationField.interface) {
+                            const validateRelationshipExistance = relationField.interface.implementations?.map(singleCardinalityValidationTemplate) as string[];
```

Comment:
It would be nice if we could do a check to ensure `relationField.interface.implementations` isn't undefined instead of relying on the `as string[]` type assertion.